frame/nfts/src/lib.rs

```diff
@@ -431,33 +437,44 @@ pub mod pallet {
 			seller: T::AccountId,
 			buyer: T::AccountId,
 		},
-		/// An `item` swap intent was created.
-		SwapCreated {
+		/// A tip was sent.
+		TipSent {
 			collection: T::CollectionId,
 			item: T::ItemId,
+			sender: T::AccountId,
+			receiver: T::AccountId,
+			amount: DepositBalanceOf<T, I>,
+		},
+		/// An `item` swap intent was created.
+		SwapCreated {
+			offered_collection: T::CollectionId,
+			offered_item: T::ItemId,
 			desired_collection: T::CollectionId,
-			desired_item: T::ItemId,
+			desired_item: Option<T::ItemId>,
 			price: Option<ItemPrice<T, I>>,
```

Comment:
Can't help but wonder if the direction has to be included with the price in some sort of wrapper.
I foresee pain dealing with two optionals that should both be either Some or None.

Reply:
done